File: 인범/2026_05_07/network.py
def solution(n, computers):
    from collections import deque
    visited = [False]*n
    answer = 0
    
    def dfs(idx,link):
        visited[idx]=True
        for i in range(len(link)):
            if link[i] == 1 and not visited[i]:
                dfs(i,computers[i])
    
    for i in range(n):
        if visited[i] == False:
            answer += 1
            dfs(i,computers[i])
        
    
    return answer

File: 인범/2026_05_07/network.java
class Solution {
    public void dfs(int idx, int[] link, boolean[] visited, int[][]computers){
        visited[idx] = true;
        for(int i=0; i<link.length; i++){
            if(!visited[i] && link[i]==1){
                dfs(i,computers[i],visited,computers);
            }
        }
    }
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[n];
        
        for(int i=0; i<n; i++){
            if(!visited[i]){
                dfs(i,computers[i],visited,computers);
                answer++;
            }
        }
        
        return answer;
    }
}